components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java

```diff
@@ -621,4 +630,193 @@ private ProxyHost getProxyConfig() {
         return proxyConfig;
     }
 
+    /**
+     * Check if the specified host is in the list of non proxy hosts.
+     *
+     * @param host          host name
+     * @param nonProxyHosts string containing the list of non proxy hosts
+     * @return true/false
+     */
+    public static boolean isHostInNonProxyList(String host, String nonProxyHosts) {
+        if ((nonProxyHosts == null) || (host == null)) {
+            return false;
+        }
+
+        /*
+         * The http.nonProxyHosts system property is a list enclosed in
+         * double quotes with items separated by a vertical bar.
+         */
+        StringTokenizer tokenizer = new StringTokenizer(nonProxyHosts, "|\"");
+
+        while (tokenizer.hasMoreTokens()) {
+            String pattern = tokenizer.nextToken();
+            if (match(pattern, host, false)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Matches a string against a pattern. The pattern contains two special
+     * characters:
+     * '*' which means zero or more characters,
+     *
+     * @param pattern         the (non-null) pattern to match against
+     * @param str             the (non-null) string that must be matched against the
+     *                        pattern
+     * @param isCaseSensitive
+     * @return <code>true</code> when the string matches against the pattern,
+     *         <code>false</code> otherwise.
+     */
+    private static boolean match(String pattern, String str,
+                                 boolean isCaseSensitive) {
```

Comment:
shall we get this to the previous line?

Reply:
Fixed in af3ac874ea63a07e139fdd85b36eeeed3d15e410